### PR TITLE
Add SC3ML event write support

### PR DESCRIPTION
**Pull request by Brtle**
Hi,

I did an XSLT to convert quakeml 1.2 into sc3ml 0.8. I think it could be valuable to obspy so I put it in a small module. It writes sc3ml from a Catalog.

I didn't do the read module but it shouldn't be too hard to add it since an XSLT is already given (http://geofon.gfz-potsdam.de/schema/0.8/).

I'm using the QuakeML unit test data but there are still 126Ko of data. Should I remove some of them?

### CODE REVIEW

**Comment by Brtle** [comment]
Trigger docs build +DOCS

**Comment by megies** [comment]
This looks really solid already, nice work! Feel free to add yourself to `CONTRIBUTORS.txt`.

File: obspy/io/seiscomp/tests/test_event.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+from obspy.io.quakeml.core import _read_quakeml
+
+
+class EventTestCase(unittest.TestCase):
+    """
+    Test suite for obspy.io.seiscomp.event
+    """
+    def setUp(self):
+        # directory where the test files are located
+        quakeml_directory = \
```
Comment (krischer) [inline obspy/io/seiscomp/tests/test_event.py]: This variable name is a bit misleading as its just the io root directory.

File: obspy/io/seiscomp/event.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    """
+    Write a SC3ML file. Since a XSLT file is used to write the SC3ML file from
+    a QuakeML file, the catalog is first converted in QuakeML.
+
+    .. warning::
+        This function should NOT be called directly, it registers via the
+        the :meth:`~obspy.core.event.Catalog.write` method of an
+        ObsPy :class:`~obspy.core.event.Catalog` object, call this instead.
+
+    :type catalog: :class:`~obspy.core.stream.Catalog`
```
Comment (krischer) [inline obspy/io/seiscomp/event.py]: This needs to be `~obspy.core.event.catalog.Catalog`.

**Comment by krischer** [comment]
AppVeyor failure is unrelated and fixed in #1636.

File: obspy/io/seiscomp/event.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    # Open filehandler or use an existing file like object.
+    if not hasattr(filename, "write"):
+        file_opened = True
+        fh = open(filename, "wb")
+    else:
+        file_opened = False
+        fh = filename
+
+    fh.write(sc3ml_doc)
```
Comment (QuLogic) [inline obspy/io/seiscomp/event.py]: Should be in a `try` with the closing in a `finally`.

File: obspy/io/seiscomp/event.py (hunk 2 above)
Comment (Brtle) [inline obspy/io/seiscomp/event.py]: Ups, I copy-pasted it from the QuakeML module. :-/ I fixed it in both modules.

File: obspy/io/seiscomp/event.py (hunk 3 above)
Comment (Brtle) [inline obspy/io/seiscomp/event.py]: I also fixed the QuakeML module.

**Comment by Brtle** [comment]
> This looks really solid already, nice work! Feel free to add yourself to CONTRIBUTORS.txt.

@megies already done ;-)

**Comment by krischer** [comment]
Thanks a bunch @Brtle!

**Pull request merged**